cc/mcts_node.cc

```diff
@@ -125,11 +125,24 @@ MctsNode::MctsNode(MctsNode* parent, Coord move)
   }
 }
 
-Coord MctsNode::GetMostVisitedMove() const {
+Coord MctsNode::GetMostVisitedMove(bool restrict_in_bensons) const {
   // Find the set of moves with the largest N.
   inline_vector<Coord, kNumMoves> moves;
+  std::array<Color, kN * kN> out_of_bounds;
+
+  if (restrict_in_bensons) {
+    out_of_bounds = position.CalculatePassAliveRegions();
+  } else {
+    for (auto& x : out_of_bounds) {
+      x = Color::kEmpty;
+    }
+  }
+
   int best_N = -1;
   for (int i = 0; i < kNumMoves; ++i) {
+    if (out_of_bounds[i] != Color::kEmpty) {
+      continue;
+    }
```

Comment:
I'm concerned that `GetMostVisitedMove(true)` could plausibly select a move with 0 visits, which might cause us to `CHECK` fail later. I can also imagine no moves being picked at all and the code `MG_CHECK(!moves.empty())` below failing.

Reply:
It's even worse/better than that, as pruning things can leave you with weird target distributions, causing training to go into NaN-land.

I'll see if i can come up with some breaking examples and get some better tests.  Shelving for now.

Reply:
oof :(